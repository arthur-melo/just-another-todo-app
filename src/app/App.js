import { useSelector } from 'react-redux';

import WelcomeCardCTA from '../features/todos/WelcomeCardCTA/WelcomeCardCTA';
import { selectTodoItems } from '../features/todos/todosSlice';
import TodoList from '../features/todos/TodoList/TodoList';
import LocalStorage from '../components/LocalStorage/LocalStorage';
import Layout from '../components/Layout/Layout';

const App = () => {
  const todoItems = useSelector(selectTodoItems);

  return (
    <LocalStorage>
      <Layout>
        <WelcomeCardCTA />
        {todoItems.length > 0 && <TodoList />}
      </Layout>
    </LocalStorage>
  );
};

export default App;
