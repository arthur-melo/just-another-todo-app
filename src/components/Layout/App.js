import React from 'react';

import './App.css';

import Header from '../Header/Header';
import ContentContainer from '../../containers/ContentContainer';
import Footer from '../Footer/Footer';

const App = () => (
  <div className="app__container">
    <Header className="app__header" />
    <ContentContainer className="app__content" />
    <Footer className="app__footer" />
  </div>
);

export default App;
