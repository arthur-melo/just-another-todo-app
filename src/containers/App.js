import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { default as AppLayout } from '../components/Layout/App';

import { AddItem } from '../actions/AddItem';
import { CancelEditItem } from '../actions/CancelEditItem';
import { DeleteItem } from '../actions/DeleteItem';
import { EditItem } from '../actions/EditItem';
import { ItemCompletion } from '../actions/ItemCompletion';
import { SelectEditItem } from '../actions/SelectEditItem';
import { LoadStateLocalStorage } from '../actions/LoadStateLocalStorage';
import { SaveStateLocalStorage } from '../actions/SaveStateLocalStorage';
import { ReorderItem } from '../actions/ReorderItem';

const appPropTypes = {
  handleAddItem: PropTypes.func.isRequired,
  handleCancelEditItem: PropTypes.func.isRequired,
  handleDeleteItem: PropTypes.func.isRequired,
  handleEditItem: PropTypes.func.isRequired,
  handleItemCompletion: PropTypes.func.isRequired,
  handleSelectEditItem: PropTypes.func.isRequired,
  handleLoadStateLocalStorage: PropTypes.func.isRequired,
  handleSaveStateLocalStorage: PropTypes.func.isRequired,
  handleReorderItem: PropTypes.func.isRequired,
};

class App extends Component {
  componentDidMount = () => this.props.handleLoadStateLocalStorage();
  componentDidUpdate = () => this.props.handleSaveStateLocalStorage(this.props.items);

  handleAddItem = itemValue => this.props.handleAddItem(itemValue);
  handleCancelEditItem = () => this.props.handleCancelEditItem();
  handleDeleteItem = selectedItemId => this.props.handleDeleteItem(selectedItemId);
  handleEditItem = modifiedItem => this.props.handleEditItem(modifiedItem);
  handleItemCompletion = modifiedItem => this.props.handleItemCompletion(modifiedItem);
  handleSelectEditItem = id => this.props.handleSelectEditItem(id);
  handleReorderItem = (initialPosition, newPosition) =>
    this.props.handleReorderItem(initialPosition, newPosition);

  render() {
    return <AppLayout {...this.props} />;
  }
}

const mapStateToProps = state => ({
  items: state.todos.items,
  editingItem: state.todos.editingItem,
});

const mapDispatchToProps = {
  handleAddItem: AddItem,
  handleCancelEditItem: CancelEditItem,
  handleDeleteItem: DeleteItem,
  handleEditItem: EditItem,
  handleItemCompletion: ItemCompletion,
  handleSelectEditItem: SelectEditItem,
  handleSaveStateLocalStorage: SaveStateLocalStorage,
  handleLoadStateLocalStorage: LoadStateLocalStorage,
  handleReorderItem: ReorderItem,
};

App.propTypes = appPropTypes;

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(App);
