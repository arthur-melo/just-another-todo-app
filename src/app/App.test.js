import renderWithProviders from '../utils/testHelpers';
import initialState from '../features/initialState';

import App from './App';

describe('App', () => {
  it('should render the App component', () => {
    const view = renderWithProviders(<App />);
    expect(view).toMatchSnapshot();
  });

  it('should render the App component with preloaded items', () => {
    const testInitialState = {
      ...initialState,
      items: [
        {
          value: 'Test todo item',
          id: '0',
          completed: false,
        },
      ],
    };

    const view = renderWithProviders(<App />, {
      preloadedState: {
        todos: testInitialState,
      },
    });

    expect(view).toMatchSnapshot();
  });
});
