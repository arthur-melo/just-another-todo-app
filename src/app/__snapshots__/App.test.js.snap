// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout__container"
      >
        <header
          class="layout__header"
        >
          <div
            class="header__container"
          >
            <nav
              class="navbar"
            >
              <span
                class="header__span navbar-brand"
              >
                <svg
                  alt="List emoji"
                  aria-hidden="true"
                  class="svg-inline--fa fa-rectangle-list fa-lg d-inline-block"
                  data-icon="rectangle-list"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 96C0 60.7 28.7 32 64 32H512c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96zM128 288c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm32-128c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zM128 384c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm96-248c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <span
                class="header__span"
              >
                Just Another Todo App
              </span>
            </nav>
          </div>
        </header>
        <main
          class="layout__content"
        >
          <div
            class="container"
          >
            <div
              class="row justify-content-center"
            >
              <div
                class="col-auto"
              >
                <div
                  class="card text-center border border-0"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Welcome!
                    </h5>
                    <p
                      class="card-text"
                    >
                      To get started, add some items to your list:
                    </p>
                    <div
                      class="container-fluid"
                    >
                      <div
                        class="row"
                      >
                        <div
                          class="mx-auto"
                        >
                          <form
                            autocomplete="on"
                            method="POST"
                          >
                            <div
                              class="d-flex"
                            >
                              <textarea
                                aria-label="Todo item description"
                                class="form-control create_todo_item__textarea"
                                id="new-todo-item"
                                name="new-todo-item"
                                placeholder="I want to do..."
                                type="text"
                              />
                              <div
                                class="mx-2"
                              >
                                <button
                                  aria-label="Add todo item"
                                  class="btn btn-primary"
                                  disabled=""
                                  title="Add todo item"
                                  type="submit"
                                >
                                  <svg
                                    aria-hidden="true"
                                    class="svg-inline--fa fa-plus fa-fw "
                                    data-icon="plus"
                                    data-prefix="fas"
                                    focusable="false"
                                    role="img"
                                    viewBox="0 0 448 512"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                                      fill="currentColor"
                                    />
                                  </svg>
                                </button>
                              </div>
                            </div>
                          </form>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </main>
        <footer
          class="layout__footer"
        >
          <div
            class="footer__container"
          >
            <p
              class="footer__p"
            >
              Offline Todo List Application â€” made with
               by
               
              <a
                href="https://github.com/arthur-melo"
                rel="noopener noreferrer"
                target="_blank"
              >
                @arthur-melo
              </a>
            </p>
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout__container"
    >
      <header
        class="layout__header"
      >
        <div
          class="header__container"
        >
          <nav
            class="navbar"
          >
            <span
              class="header__span navbar-brand"
            >
              <svg
                alt="List emoji"
                aria-hidden="true"
                class="svg-inline--fa fa-rectangle-list fa-lg d-inline-block"
                data-icon="rectangle-list"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 96C0 60.7 28.7 32 64 32H512c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96zM128 288c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm32-128c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zM128 384c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm96-248c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span
              class="header__span"
            >
              Just Another Todo App
            </span>
          </nav>
        </div>
      </header>
      <main
        class="layout__content"
      >
        <div
          class="container"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-auto"
            >
              <div
                class="card text-center border border-0"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Welcome!
                  </h5>
                  <p
                    class="card-text"
                  >
                    To get started, add some items to your list:
                  </p>
                  <div
                    class="container-fluid"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="mx-auto"
                      >
                        <form
                          autocomplete="on"
                          method="POST"
                        >
                          <div
                            class="d-flex"
                          >
                            <textarea
                              aria-label="Todo item description"
                              class="form-control create_todo_item__textarea"
                              id="new-todo-item"
                              name="new-todo-item"
                              placeholder="I want to do..."
                              type="text"
                            />
                            <div
                              class="mx-2"
                            >
                              <button
                                aria-label="Add todo item"
                                class="btn btn-primary"
                                disabled=""
                                title="Add todo item"
                                type="submit"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="svg-inline--fa fa-plus fa-fw "
                                  data-icon="plus"
                                  data-prefix="fas"
                                  focusable="false"
                                  role="img"
                                  viewBox="0 0 448 512"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                                    fill="currentColor"
                                  />
                                </svg>
                              </button>
                            </div>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
      <footer
        class="layout__footer"
      >
        <div
          class="footer__container"
        >
          <p
            class="footer__p"
          >
            Offline Todo List Application â€” made with
             by
             
            <a
              href="https://github.com/arthur-melo"
              rel="noopener noreferrer"
              target="_blank"
            >
              @arthur-melo
            </a>
          </p>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`App should render the App component with preloaded items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout__container"
      >
        <header
          class="layout__header"
        >
          <div
            class="header__container"
          >
            <nav
              class="navbar"
            >
              <span
                class="header__span navbar-brand"
              >
                <svg
                  alt="List emoji"
                  aria-hidden="true"
                  class="svg-inline--fa fa-rectangle-list fa-lg d-inline-block"
                  data-icon="rectangle-list"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 96C0 60.7 28.7 32 64 32H512c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96zM128 288c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm32-128c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zM128 384c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm96-248c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <span
                class="header__span"
              >
                Just Another Todo App
              </span>
            </nav>
          </div>
        </header>
        <main
          class="layout__content"
        >
          <div
            class="container"
          >
            <div
              class="row justify-content-center"
            >
              <div
                class="col-auto"
              >
                <div
                  class="card text-center border border-0"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Welcome!
                    </h5>
                    <p
                      class="card-text"
                    >
                      To get started, add some items to your list:
                    </p>
                    <div
                      class="container-fluid"
                    >
                      <div
                        class="row"
                      >
                        <div
                          class="mx-auto"
                        >
                          <form
                            autocomplete="on"
                            method="POST"
                          >
                            <div
                              class="d-flex"
                            >
                              <textarea
                                aria-label="Todo item description"
                                class="form-control create_todo_item__textarea"
                                id="new-todo-item"
                                name="new-todo-item"
                                placeholder="I want to do..."
                                type="text"
                              />
                              <div
                                class="mx-2"
                              >
                                <button
                                  aria-label="Add todo item"
                                  class="btn btn-primary"
                                  disabled=""
                                  title="Add todo item"
                                  type="submit"
                                >
                                  <svg
                                    aria-hidden="true"
                                    class="svg-inline--fa fa-plus fa-fw "
                                    data-icon="plus"
                                    data-prefix="fas"
                                    focusable="false"
                                    role="img"
                                    viewBox="0 0 448 512"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                                      fill="currentColor"
                                    />
                                  </svg>
                                </button>
                              </div>
                            </div>
                          </form>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="row justify-content-center"
            >
              <div
                class="col-12 col-md-9 col-lg-7 col-xl-6"
              >
                <div
                  class="todo_list__todos"
                >
                  <ul
                    class="list-group"
                  >
                    <div>
                      <div
                        aria-describedby="rfd-hidden-text-0-hidden-text-0"
                        class="todo_list__todos__li"
                        data-rfd-drag-handle-context-id="0"
                        data-rfd-drag-handle-draggable-id="0"
                        data-rfd-draggable-context-id="0"
                        data-rfd-draggable-id="0"
                        data-testid="todo-list-item"
                        draggable="false"
                        role="button"
                        tabindex="0"
                      >
                        <li
                          class="todo_item__component list-group-item form-control d-flex border border-0  align-items-center"
                          data-testid="todo_item-listitem"
                        >
                          <div
                            class="todo_item__checkout"
                            data-testid="todo_item-item-completion"
                          >
                            <span>
                              <svg
                                aria-hidden="true"
                                class="svg-inline--fa fa-square fa-fw "
                                data-icon="square"
                                data-prefix="fas"
                                focusable="false"
                                font-size="36px"
                                role="img"
                                viewBox="0 0 448 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M0 96C0 60.7 28.7 32 64 32H384c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96z"
                                  fill="currentColor"
                                />
                              </svg>
                            </span>
                          </div>
                          <div
                            class="border rounded w-100 form-control todo_item__text text-break"
                          >
                            Test todo item
                          </div>
                        </li>
                      </div>
                    </div>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </main>
        <footer
          class="layout__footer"
        >
          <div
            class="footer__container"
          >
            <p
              class="footer__p"
            >
              Offline Todo List Application â€” made with
               by
               
              <a
                href="https://github.com/arthur-melo"
                rel="noopener noreferrer"
                target="_blank"
              >
                @arthur-melo
              </a>
            </p>
          </div>
        </footer>
      </div>
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rfd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rfd-hidden-text-0-hidden-text-0"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div
      class="layout__container"
    >
      <header
        class="layout__header"
      >
        <div
          class="header__container"
        >
          <nav
            class="navbar"
          >
            <span
              class="header__span navbar-brand"
            >
              <svg
                alt="List emoji"
                aria-hidden="true"
                class="svg-inline--fa fa-rectangle-list fa-lg d-inline-block"
                data-icon="rectangle-list"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 96C0 60.7 28.7 32 64 32H512c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96zM128 288c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm32-128c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zM128 384c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm96-248c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span
              class="header__span"
            >
              Just Another Todo App
            </span>
          </nav>
        </div>
      </header>
      <main
        class="layout__content"
      >
        <div
          class="container"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-auto"
            >
              <div
                class="card text-center border border-0"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Welcome!
                  </h5>
                  <p
                    class="card-text"
                  >
                    To get started, add some items to your list:
                  </p>
                  <div
                    class="container-fluid"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="mx-auto"
                      >
                        <form
                          autocomplete="on"
                          method="POST"
                        >
                          <div
                            class="d-flex"
                          >
                            <textarea
                              aria-label="Todo item description"
                              class="form-control create_todo_item__textarea"
                              id="new-todo-item"
                              name="new-todo-item"
                              placeholder="I want to do..."
                              type="text"
                            />
                            <div
                              class="mx-2"
                            >
                              <button
                                aria-label="Add todo item"
                                class="btn btn-primary"
                                disabled=""
                                title="Add todo item"
                                type="submit"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="svg-inline--fa fa-plus fa-fw "
                                  data-icon="plus"
                                  data-prefix="fas"
                                  focusable="false"
                                  role="img"
                                  viewBox="0 0 448 512"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                                    fill="currentColor"
                                  />
                                </svg>
                              </button>
                            </div>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-12 col-md-9 col-lg-7 col-xl-6"
            >
              <div
                class="todo_list__todos"
              >
                <ul
                  class="list-group"
                >
                  <div>
                    <div
                      aria-describedby="rfd-hidden-text-0-hidden-text-0"
                      class="todo_list__todos__li"
                      data-rfd-drag-handle-context-id="0"
                      data-rfd-drag-handle-draggable-id="0"
                      data-rfd-draggable-context-id="0"
                      data-rfd-draggable-id="0"
                      data-testid="todo-list-item"
                      draggable="false"
                      role="button"
                      tabindex="0"
                    >
                      <li
                        class="todo_item__component list-group-item form-control d-flex border border-0  align-items-center"
                        data-testid="todo_item-listitem"
                      >
                        <div
                          class="todo_item__checkout"
                          data-testid="todo_item-item-completion"
                        >
                          <span>
                            <svg
                              aria-hidden="true"
                              class="svg-inline--fa fa-square fa-fw "
                              data-icon="square"
                              data-prefix="fas"
                              focusable="false"
                              font-size="36px"
                              role="img"
                              viewBox="0 0 448 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M0 96C0 60.7 28.7 32 64 32H384c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96z"
                                fill="currentColor"
                              />
                            </svg>
                          </span>
                        </div>
                        <div
                          class="border rounded w-100 form-control todo_item__text text-break"
                        >
                          Test todo item
                        </div>
                      </li>
                    </div>
                  </div>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </main>
      <footer
        class="layout__footer"
      >
        <div
          class="footer__container"
        >
          <p
            class="footer__p"
          >
            Offline Todo List Application â€” made with
             by
             
            <a
              href="https://github.com/arthur-melo"
              rel="noopener noreferrer"
              target="_blank"
            >
              @arthur-melo
            </a>
          </p>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
